qcodes/instrument_drivers/oxford/IPS120.py

```diff
@@ -0,0 +1,944 @@
+        self.remote_status(2)
+
+    def close(self):
+        """Safely close connection"""
+        logging.info(__name__ + ' : Closing IPS120 connection')
+        self.local()
+        super().close()
+
+    def _get_remote_status(self):
+        """
+        Get remote control status
+
+        Returns:
+            result(str) :
+            "Local & locked",
+            "Remote & locked",
+            "Local & unlocked",
+            "Remote & unlocked",
+            "Auto-run-down",
+            "Auto-run-down",
+            "Auto-run-down",
+            "Auto-run-down"
+        """
+        logging.info(__name__ + ' : Get remote control status')
+        result = self._execute('X')
+        val_mapping = {0: "Local and locked",
+                       1: "Remote and locked",
+                       2: "Local and unlocked",
+                       3: "Remote and unlocked",
+                       4: "Auto-run-down",
+                       5: "Auto-run-down",
+                       6: "Auto-run-down",
+                       7: "Auto-run-down"}
+        return val_mapping[int(result[6])]
+
+    def _set_remote_status(self, mode):
+        """
+        Set remote control status.
+
+        Args:
+            mode(int) :
+            0 : "Local and locked",
+            1 : "Remote and locked" (not available),
+            2 : "Local and unlocked",
+            3 : "Remote and unlocked"
+        """
+        status = {
+            0: "Local and locked",
+            2: "Local and unlocked",
+            3: "Remote and unlocked",
+        }
+        if status.__contains__(mode):
+            logging.info(
+                __name__ +
+                ' : Setting remote control status to %s' %
+                status.get(
+                    mode,
+                    "Unknown"))
+            self._execute('C%s' % mode)
+        else:
+            print('Invalid mode inserted: %s' % mode)
+
+    def _get_system_status(self):
+        """
+        Get the system status
+
+        Returns:
+            result (str) :
+            "Normal",
+            "Quenched",
+            "Over Heated",
+            "Warming Up",
+            "Fault"
+        """
+        result = self._execute('X')
+        logging.info(__name__ + ' : Getting system status')
+        status = {0: "Normal",
+                  1: "Quenched",
+                  2: "Over Heated",
+                  3: "Warming Up",
+                  4: "Fault"}
+        return status[int(result[1])]
+
+    def _get_system_status2(self):
+        """
+        Get the system status
+
+        Returns:
+            result (str) :
+            "Normal",
+            "On positive voltage limit",
+            "On negative voltage limit",
+            "Outside negative current limit",
+            "Outside positive current limit"
+        """
+        result = self._execute('X')
+        logging.info(__name__ + ' : Getting system status')
```

Comment:
It might be worth replacing the logger with a pattern like 
```
log = logging.getLogger(__name__)
```
once and then after that 
```
log.warning('this is a warning`)
```
such as in 

https://github.com/QCoDeS/Qcodes/blob/master/qcodes/instrument_drivers/tektronix/AWG5014.py#L15